internal/app/machined/pkg/controllers/network/etcfile.go

```diff
@@ -214,3 +246,47 @@ func (ctrl *EtcFileController) renderHosts(hostnameStatus *network.HostnameStatu
 
 	return buf.Bytes(), nil
 }
+
+func atomicFileWrite(dst string, b []byte, perm os.FileMode) error {
```

Comment:
we seem to be creating too many functions doing same thing. in this case, as we mount this file into the container, I think renaming won't work with bind mounting properly.

it's better to use here a write which doesn't rewrite the file if it doesn't change. 

See https://github.com/siderolabs/talos/pull/8126/files#diff-025f2ce8630bc16d48cc048303aa0fe8e6b42d5f5ee59056c59163fbcc0ff2deR69-R78